rust/ql/test/extractor-tests/macro_expansion/macro_expansion.rs

```diff
@@ -0,0 +1,2 @@
+#[ctor::ctor]
+fn foo() {}
```

Comment:
Could you add some more extensive tests?  Covering items with more than one macro attribute, and also some macro attributes that expand to "nothing", like `#[cfg(all(unix,  not(unix)))]` or perhaps even a macro attribute that adds another macro attribute?

Reply:
in order to do that we would need to define our own attribute macro. That requires a separate crate, so it will probably require some additional work on the qltest side to pull off. Trying something that expand to nothing is easy though.

Reply:
If a custom attribute macro in the test requires tweaking the qltest code, I would very much prefer to do that in a follow-up PR

Reply:
btw, I'm not even sure `cfg` is really implemented as an attribute macro, even if it behaves like one. There's definitely no instance of regular attribute macro definitions in the standard library, this I checked (I would have used those instead of relying on a third party crate).

Reply:
yeah, I can confirm `cfg` does not work through attribute macro expansion

Reply:
I've added an additional integration test (to be able to use an additional crate for the macros) covering the cases you mentioned (using a `repeat` attribute macro).